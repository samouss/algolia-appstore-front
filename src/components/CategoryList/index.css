.CategoryList {
  background-color: #ffffff;
  border: 1px solid #f2f2f2;
  min-height: 272px;
  width: 210px;

  margin-left: 1vh;
  &:first-child {
    margin-left: 0;
  }
}
