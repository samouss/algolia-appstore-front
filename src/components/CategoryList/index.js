import React from 'react';
import PropTypes from 'prop-types';
import FilterList from 'components/FilterList';
import FilterItem from 'components/FilterItem';
import Counter from 'components/Counter';
import style from './index.css';

const CategoryList = ({ facetValues, onChange }) => (
  <FilterList
    customClassName={style.CategoryList}
  >
    {facetValues.map(({ name, count, isRefined }) => (
      <FilterItem
        key={name}
        value={name}
        isRefined={isRefined}
        onChange={() => onChange(name, isRefined)}
      >
        <span>{name}</span>

        <Counter
          value={count}
        />
      </FilterItem>
    ))}
  </FilterList>
);

CategoryList.propTypes = {
  facetValues: PropTypes.arrayOf(PropTypes.shape({
    name: PropTypes.string.isRequired,
    count: PropTypes.number.isRequired,
    isRefined: PropTypes.bool.isRequired,
  })).isRequired,
  onChange: PropTypes.func.isRequired,
};

export default CategoryList;
