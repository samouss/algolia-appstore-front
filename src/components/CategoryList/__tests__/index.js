import React from 'react';
import { shallow } from 'enzyme';
import CategoryList from '../index';

describe('<CategoryList />', () => {
  const defaultProps = {
    facetValues: [
      {
        name: 'Games',
        count: 1234,
        isRefined: true,
      },
      {
        name: 'Social',
        count: 5678,
        isRefined: false,
      },
    ],
    onChange: () => {},
  };

  it('expect to render', () => {
    const props = {
      ...defaultProps,
    };

    const component = shallow(
      <CategoryList
        {...props}
      />,
    );

    expect(component).toMatchSnapshot();
  });

  it('expect to call onChange', () => {
    const props = {
      ...defaultProps,
      onChange: jest.fn(),
    };

    const component = shallow(
      <CategoryList
        {...props}
      />,
    );

    component
      .find('CategoryItem')
      .first()
      .simulate('change');

    expect(props.onChange).toHaveBeenCalled();
  });
});
