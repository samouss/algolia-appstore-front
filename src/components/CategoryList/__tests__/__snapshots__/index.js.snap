// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`<CategoryList /> expect to render 1`] = `
<FilterList
  customClassName="CategoryList"
>
  <FilterItem
    isRefined={true}
    onChange={[Function]}
  >
    <span>
      Games
    </span>
    <Counter
      customClassName=""
      value={1234}
    />
  </FilterItem>
  <FilterItem
    isRefined={false}
    onChange={[Function]}
  >
    <span>
      Social
    </span>
    <Counter
      customClassName=""
      value={5678}
    />
  </FilterItem>
</FilterList>
`;
